### PR TITLE
ab: update compiler.py

### PR DESCRIPTION
**Pull request by annagav**

### CODE REVIEW

**Comment by pdpinch** [comment]
I'm not sure how we should review this?

**Comment by annagav** [comment]
Now you should be able to click on preview in the dashboard and see the assignments page.

**Comment by ald32** [comment]
@annagav , can I see this yet?

**Comment by annagav** [comment]
@ald32, you cannot see this yet, it is still under review.

File: instructor/compiler.py [hunk 1]
```diff
@@ -347,17 +337,17 @@ def primary_anti_body(assignment):
         for band in a.bands.all():
             if western_blot.has_whole_cell_lysate:
```
Comment (noisecapella) [inline instructor/compiler.py]: Is it possible to combine these if statements? Is it possible for `western_blot` to have more than one `has_` flag set to true at a time?

File: html_app/assignments/scb_ex/addmutlpledialog.soy [hunk 2]
```diff
@@ -1,93 +0,0 @@
-{namespace scb_ex_old}
-
-/**
-@param template
-@param assignment
-@param experiment
-*/
-{template .dialog}
-<div class='scb_ex_inner_dialog' role='dialog' aria-label='Add Samples'>
```
Comment (noisecapella) [inline html_app/assignments/scb_ex/addmutlpledialog.soy:9]: Is there an easy way to view this dialog box somewhere? I'm assuming this was a duplicate of the `{template .dialog}` in the file above but there are minor differences

File: instructor/compiler.py [hunk 3]
```diff
@@ -372,56 +362,49 @@ def primary_anti_body(assignment):
...
-        protocol = strain_protocol.protocol
-        for t in protocol.treatments.all():
-            tr = t.concentration
-            ret[str(tr)] = {
-                'name': str(tr),
-                'value': tr
-            }
+    for strain_protocol in assignment.strain_treatment.filter(enabled=True):
+        treatment = strain_protocol.treatment
+        tr = treatment.drug.concentration
```
Comment (noisecapella) [inline instructor/compiler.py]: Can you rename `tr` to `concentration`?

File: instructor/compiler.py [hunk 4]
```diff
@@ -372,56 +362,49 @@ def primary_anti_body(assignment):
...
+        ret[str(tr)] = {
```
Comment (noisecapella) [inline instructor/compiler.py]: It seems like `str()` is unnecessary here since `concentration` is a `CharField`

File: instructor/compiler.py [hunk 5]
```diff
@@ -372,56 +362,49 @@ def primary_anti_body(assignment):
...
-        protocol = strain_protocol.protocol
-        for t in protocol.treatments.all():
-            tr = t.temperature
-            ret[str(tr)] = {
-                'name': str(tr),
-                'value': tr
-            }
+    for strain_protocol in assignment.strain_treatment.filter(enabled=True):
+        treatment = strain_protocol.treatment
+        t = treatment.temperature
```
Comment (noisecapella) [inline instructor/compiler.py]: Can you rename `t` and `tr` to `temperature` and `degrees` respectively?

File: instructor/compiler.py (hunk 1 above)
Comment (annagav) [inline instructor/compiler.py]: I honestly think this 'marks' list is not needed at all. It is useful only if I want to set a band for this particular antibody to appear for all samples. I don't think such case is common, and the assignment builder does not provide such option to the instructor.

File: html_app/assignments/scb_ex/addmutlpledialog.soy (hunk 2 above)
Comment (annagav) [inline html_app/assignments/scb_ex/addmutlpledialog.soy:9]: If you change the namespace here to scb_ex and in addmultipledioalog2.soy change the namespace to something else, and go to setup any of the NSF sample exercises, this older version of the dialog will be used.

**Comment by noisecapella** [comment]
:+1:

**Pull request merged**